fatiando/gridder.py

```diff
@@ -439,3 +444,269 @@ def cut(x, y, scalars, area):
+    * xy : N-D array (optional)
+        [MxN] array where M is the number of observation points and N is the 
+        dimension.  This is effectively a concatenated xp,yp, etc...
+    * npd : tuple (optional)
+        Optional tuple containing the total number of desired elements in each
+        dimension
+    * padtype : string (optional)
+        String describing what to pad the new values with. Options:
+
+        [ OddReflectionTaper | OddReflection | Reflection | value | LinTaper
+        | edge | mean ]
+            OddReflectionTaper - Generates odd reflection then tapers to the 
+            mean using a cosine function (Default)
+
+            OddReflection - Pads with the odd reflection, with no taper
+
+            Reflection - Pads with simple reflection
+
+            LinTaper - Linearly tapers to the mean
+
+            value - numeric value
+
+            edge - uses the edge value as a constant pad
+
+            mean - uses the mean of the vector along each axis
+
+    Returns:
+
+    * ap : numpy array
+        Padded array. The array core is a deep copy of the original array
+    * cp : list
+        List of coordinate arrays containing the extrapolated coordinate values
+        Returns None if xy not provided
+    * nps : list
+        List of tuples containing the number of elements padded onto each 
+        dimension.
+
+    """
+
+    # Test to make sure padtype is valid
+    padopts = ['','oddreflectiontaper','oddreflection', 'reflection',
+            'lintaper', 'edge', 'value', 'mean']
+    #API Note:
+    # If one wishes to add more options to the padding, use the following
+    # checklist to make sure all the sections are consistent.
+    #  [ ] Docstring (2 spots: the options, and the option descriptions
+    #  [ ] padopts - Make sure to add to avoid raising a ValueError
+    #  [ ] if statements below - Finally add an elif to the if statements
+    #                            below. Add a descriptive comment.
+    if padtype.lower() not in padopts and not _is_number(padtype):
+        raise ValueError('gridder.pad_array: Pad Type not understood')
+    # If npd is not provided, populate with next power of 2
+    npt = []
+    nd = a.ndim
+    if npd == None:
+        for ii in range(0, nd):
+            if nd == 1:
+                npt.append(_nextpow2(len(a)))
+            else:
+                npt.append(_nextpow2(a.shape[ii]))
+    else:
+        et = 'gridder.pad_array: pad dimensions do not match array dims'
+        if nd == 1:
+            if _is_integer(npd):
+                npt.append(npd)
+            else:
+                if len(npd) != 1:
+                    raise ValueError(et)
+                else:
+                    npt.append(npd)
+        else:
+            if _is_integer(npd):
+                raise ValueError(et)
+            elif len(npd) != a.ndim:
+                raise ValueError(et)
+            else:
+                npt = npd
+        for ii in numpy.arange(len(npt)):
```

Comment:
Be careful of `numpy.arange` for looping over an index because it  might not produce integers (that would break the indexing code below). It's always better to use `range` or `xrange`.